docs/DeveloperGuide.md

```diff
@@ -1,34 +1,726 @@
+  different location or with a different expiry date. This is not allowed as of v2.0.
+  * Error Message: `"Sorry my friend, you have added this food before but in a different location or have different expiry dates.
+      Please specify another foodname."`
+
+_Exceptions related to file storage:_
+
+* **`StorageLoadingException`**: Is thrown when an error occurred during loading of the saved data.
+  * Error Message: `"There was an error loading the data for FridgeFriend!"`
+* **`StorageSavingException`**: Is thrown when an error occurred during saving of the current data.
+  * Error Message: `"There was an error saving the data for FridgeFriend!"`
+
+## Implementation
+
+This section covers the application flow and more specific interactions between classes and their methods. A overview of the main application logic would be covered first, before addressing several specific application commands.
+
+### Main Logic
+
+The main application logic underlines the main executable conditions that make up the bulk of the application. Below is the outline of the logic:
+
+1. `FridgeFriend` uses the `Ui` class to obtain the user input.
+2. `FridgeFriend` then uses the `Parser` class to parse the user input.
+3. This results in the construction of a `Command` object that is returned to `FridgeFriend`.
+4. Following that, `FridgeFriend` then self-invokes a method that executes the `Command` object.
+5. In this method, the `Command` object  executes its specified action.
+6. From steps 2 to 5, should there be an error thrown, the `Command` class would construct an `Exception` object.
+7. Lastly, the `Exception` would be thrown back to `FridgeFriend`, which will then use the `Ui` class to display an error message to the user.
+
+Given below is the sequence diagram for the interactions within the main application logic.
+
+![MainLogicSequenceDiagram](diagrams/diagram_images/MainLogicSequenceDiagram.png)
+
+:information_source: Information:
+
+* The lifeline for `Parser`, `Command`, and `Exception` should end at the destroy marker. However, due to a limitation of PlantUML, the three lifelines reach the end of the diagram.
+* Due to the lack of a standard to represent try-catch blocks in UML, the `alt` frame in this diagram is used to indicate a try-catch block.
+  Additionally, while the diagram shows the `Exception` object being initialised by `FridgeFriend`, it is actually initialised in the `Parser` or `Command` class and thrown to `FridgeFriend`
+
+### Add Command
+
+AddCommand class is initialized whenever the parser recognize the `add` keyword.
+
+1. The constructor should create a new Food object according to the parsed user input.
+2. The UniqueFoodnameChecker will be called from Fridge class to decide whether to add a new `Food` or edit on existing `Food`.
+
+Given below is the sequence diagram for the AddCommand workflow.
+
+![AddCommandSequenceDiagram](diagrams/diagram_images/AddSequenceDiagram.png)
```

Comment:
![DG review4](https://user-images.githubusercontent.com/77476720/113237877-3edb0900-92da-11eb-9a2a-e76b9a96063b.jpg)

What does it returns over here the highlighted part?